### ISSUE TITLE
HCPCS concepts with drug dosage inconsistent mapping

### ISSUE THREAD

**Issue by dimshitc** [OHDSI/Vocabulary-v5.0#721]
https://athena.ohdsi.org/search-terms/terms/2615395
source concept stands for 'per ml' while you map to concept with exact dosage

https://athena.ohdsi.org/search-terms/terms/2718363
source concept stands for 10 mg, while target concept for 100 mg

https://athena.ohdsi.org/search-terms/terms/2718636
here, you map to the concept that doesn't have a dosage, while in April version of a vocabulary it was mapped to "pyridoxine hydrochloride 100 MG/ML Injectable Solution"

Is there any convention of how to treat the dosage of drug mentioned in HCPCS code?

**Comment by m-khitrun** [comment]
Hello, @dimshitc,
Thank you for the comment and sorry for the late reply.
Currently we map all HCPCS drugs manually. We try to find out the brand name and the dosage that correspond to the particular HCPCS code, and then map to the most granular applicable concept in RxNorm.
Correct me if I'm wrong, but in source HCPCS codes not dosages but billing units are mentioned, and numbers in source and target concepts may vary significantly. 
Our aim is to improve drug mapping in HCPCS, that's why you see the above mentioned changes.

**Comment by dimshitc** [comment]
https://athena.ohdsi.org/search-terms/terms/2718363
source concept stands for 10 mg, while target concept for 100 mg

Did you look in a source data? Does they put quanitity of 10 of J1260, when they want to say that a patient got 100 mg?

**Comment by m-khitrun** [comment]
> https://athena.ohdsi.org/search-terms/terms/2718363 source concept stands for 10 mg, while target concept for 100 mg

In the source code 10 mg is a billing unit. The target concepts describes the way the drug is supplied. None of them describes the dosage.
Actually, mapping of this particular code is not appropriate, and it will be remapped to a branded drug for the next release.

> Did you look in a source data? Does they put quanitity of 10 of J1260, when they want to say that a patient got 100 mg?

Yes, according to the coding guidelines (e.g. [here ](https://www.aao.org/practice-management/coding/injectable-drugs)) the number of billing units is multiplied according to the dose administered.

**Comment by dimshitc** [comment]
So with this approach, when doing an ETL, user needs to convert the source quantity into drug_exposure.quantity by the (target concept dosage / source concept dosage ratio)?

**Comment by dimshitc** [comment]
Actually it would make sense to look not in the coding guidelines only, but in a real data, do they put the quantity with the HCPCS codes

**Comment by dimshitc** [comment]
So, do agree with the mapping of dosages approach I suggested?
@m-khitrun @Alexdavv

**Comment by m-khitrun** [comment]
I'm afraid I have nothing to say in addition to the above-mentioned. These are two different entities: the dose administered to the patient and the dose supplied by manufacturer. We can't equate them in the course of mapping.
@cgreich, would you like to join the discussion?

**Comment by cgreich** [comment]
Yeah. We do have a problem here. Because RxNorm/Ext covers the **strength** of a product (or drug component), while 
HCPCS and the like give you the **dose** of what was administered, but not the actual product.

Right now, the mapping goes to the best concept, which is the one where strength is as close as possible to dose, if possible identical. But the best could be a factor of 10 away from the needed one, like in the example above.

We need to start thinking how to fix that, in particular that we are seeing dosing studies, like what @tiozab is doing these days.

- Create some standard for utilizing sig for the conversion between strength and dose
- Add a new field dose_factor, which is used for conversion between strength and dose
- Create a convention to use ingredient + quantity, with some mapping mechanism
- Create a pseudo product in RxNorm Extension

Let's start the conversation.

**Comment by dimshitc** [comment]
I like the 
Create a pseudo product in RxNorm Extension idea. 
it seems to be the easiest for users
